d2core/d2asset/animation_manager.go

```diff
@@ -20,10 +20,27 @@ func createAnimationManager() *animationManager {
 	return &animationManager{d2common.CreateCache(animationBudget)}
 }
 
+// Dispose of all the animations to prevent ebiten storing the images
+func (am *animationManager) ClearCache() {
+	keys := am.cache.GetKeys()
+	for idx := range keys {
+		key := keys[idx]
+
+		//TODO: move fonts in another cache ?
+		if strings.HasPrefix(key, "/data/local/FONT/") {
+			continue;
+		}
+		animation, _:= am.cache.Retrieve(key)
+		animation.(*Animation).Dispose()
+	}
+
+	am.cache.Clear()
+}
+
 func (am *animationManager) loadAnimation(animationPath, palettePath string, transparency int) (*Animation, error) {
 	cachePath := fmt.Sprintf("%s;%s;%d", animationPath, palettePath, transparency)
 	if animation, found := am.cache.Retrieve(cachePath); found {
-		return animation.(*Animation).Clone(), nil
+			return animation.(*Animation), nil
```

Comment:
Will this not make all animations of the same type point at the same object?

Keep in mind that there can be multiple instances of the same animation loaded but with different properties (speed, direction, etc).

Reply:
Wouldn't returning the dereference cause it to copy the object? The gotcha here would be if the animation object had pointers -- if it does, then this would retain the shared data erroneously.